src/views/Events/index.js

```diff
@@ -79,6 +227,7 @@ export default class Events extends Component {
       const futureEvents = gordonEvent.getFutureEvents(this.state.allEvents);
       await this.setState({ events: futureEvents });
     }
+    this.createURLParameters();
```

Comment:
I'd suggest breaking togglePastEvents() into 2 functions.  One is the toggle part, and the other is the rest, starting here.  Then you can call both, or just one.